src/main/kotlin/org/dashevo/dpp/contract/ContractFactory.kt

```diff
@@ -9,49 +9,37 @@ package org.dashevo.dpp.contract
 
 import org.dashevo.dpp.Factory
 import org.dashevo.dpp.util.Cbor
+import org.dashevo.dpp.util.Entropy
+import org.dashevo.dpp.util.HashUtils
 
 class ContractFactory : Factory() {
 
-    fun createDataContract(rawDataContract: MutableMap<String, Any?>): Contract {
+    fun createDataContract(ownerId: String, rawDataContract: MutableMap<String, Any?>): DataContract {
 
-        val contractId = if (rawDataContract.containsKey("contractId")) rawDataContract["contractId"] as String else ""
+        val dataContractEntropy = Entropy.generate()
+        val dataContractId = HashUtils.generateDataContractId(ownerId, dataContractEntropy)
 
-        val contract = Contract(contractId,
+        val dataContract = DataContract(dataContractId,
+                ownerId,
+                DataContract.SCHEMA,
                 rawDataContract["documents"] as MutableMap<String, Any?>)
 
-        if (rawDataContract.containsKey("\$schema")) {
-            contract.schema = rawDataContract["\$schema"] as String
-        }
+        dataContract.entropy = dataContractEntropy
 
-        if (rawDataContract.containsKey("version")) {
-            contract.version = rawDataContract["version"] as Int
-        }
-
-        if (rawDataContract.containsKey("definitions")) {
-            contract.definitions = rawDataContract["definitions"] as MutableMap<String, Any>
-        }
-
-        return contract
-    }
-
-    fun create(contractId: String, documents: MutableMap<String, Any?>): Contract {
-        val rawContract = HashMap<String, Any?>(2)
-        rawContract["contractId"] = contractId
-        rawContract["documents"] = documents
-        return createDataContract(rawContract)
+        return dataContract
     }
 
-    fun createFromObject(rawContract: MutableMap<String, Any?>, options: Options = Options()): Contract {
-        return createDataContract(rawContract)
+    fun createFromObject(rawContract: MutableMap<String, Any?>, options: Options = Options()): DataContract {
```

Comment:
It looks like the `options` parameter is not being used and since this method is just calling the `DataContract` constructor, can we remove it and call the constructor directly?

Reply:
The idea behind the parameters comes from the JavaScript method.  However, we don't have json schema validation in our library, so the options are empty and unused.  In JavaScript, the options are always a json object.  At this point, I am not sure about the best way to handle that in our code.  In other places where JavaScript uses `options`, I have a boolean parameter (like `skipValidation` would be in this case) instead.

```javascript
  /**
   * Create Data Contract from plain object
   *
   * @param {RawDataContract} rawDataContract
   * @param {Object} options
   * @param {boolean} [options.skipValidation=false]
   * @return {Promise<DataContract>}
   */
  async createFromObject(rawDataContract, options = { }) {
    const opts = { skipValidation: false, ...options };

    if (!opts.skipValidation) {
      const result = await this.validateDataContract(rawDataContract);

      if (!result.isValid()) {
        throw new InvalidDataContractError(result.getErrors(), rawDataContract);
      }
    }

    return new DataContract(rawDataContract);
  }
```